### DISCUSSION TITLE
Multiple Set-Cookie headers on a static response object

### DISCUSSION THREAD

**Discussion by ARBrooks** [cypress-io/cypress#16353] [Questions and Help]
Hi, I've got a scenario where I'm stubbing an endpoint that sets multiple cookies on success via multiple set-cookie headers using the intercept API. 

With a static response object, when you define a headers option it only enables you to define a single header/ set-cookie entry with the typings strictly only allowing strings, so if you have, for example:

```
{
     set-cookie: ['cookie1=1', 'cookie2=2']
}
```
Cypress throws and exception, but there's no alternative that I've seen to define the set-cookie, as a multiple with a stub. I've tried alternatives like hooking into the req/ res using the handler functions but this isn't fulfilling the upfront stub requirement as it appears to still allow the request through to the destination URL before modifying the response headers with our extra set-cookie headers. 

Does anyone know of a suitable work around that will fulfil this use case?

**Comment by tvsbrent** [comment]
I cannot provide any answer for you on this, but the error you are seeing is coming from here:

https://github.com/cypress-io/cypress/blob/8e23567d5b2b851d9302c0019ae0e62ff546604d/packages/driver/src/cy/net-stubbing/static-response-utils.ts#L41-L43

It may just be that we need to make a feature request to support array headers in static responses.

**Reply by tvsbrent** [reply]
To follow up on this more, I don't know if this would work in your use case, but I ended up using this signature for the `reply` method:

```
req.reply(statusCode, body, headers)
```

That seemed to take a header entry whose value was an array without complaining.

**Reply by yabab** [reply]
Tried your workaround @tvsbrent, but it seems to have been patched as of 9.5.2.

My QA's are currently just setting the cookies manually before the request is sent, but the way I see it their tests would still pass if something messed with the cookie setup code.
